backend/lucene/src/main/java/org/hibernate/search/backend/lucene/orchestration/impl/LuceneWriteWorkProcessor.java

```diff
@@ -81,6 +81,20 @@ void beforeWorkSet(DocumentCommitStrategy commitStrategy, DocumentRefreshStrateg
 				|| DocumentRefreshStrategy.FORCE.equals( refreshStrategy );
 	}
 
+	void ensureIndexExists() {
```

Comment:
I would add a comment over here, explaining the reason we need this method bypassing the normal submit. That would be what you explain in the pull request description, ie the method do not set hasUncommittedWorks to true, in order to not provoke a commit.